src/cli/Microsoft.DotNet.UpgradeAssistant.Cli/Commands/Upgrade/ConsoleUpgrade.cs

```diff
@@ -76,9 +86,32 @@ public async Task RunAsync(CancellationToken token)
             {
                 // Do not pass the same token as it may have been canceled and we still need to persist this.
                 await _stateManager.SaveStateAsync(context, default);
+                await WriteUpgradeReport(context, token);
             }
         }
 
+        private async Task<bool> WriteUpgradeReport(IUpgradeContext context, CancellationToken token)
+        {
+            var outputDirectory = Path.GetDirectoryName(context.InputPath);
+            if (outputDirectory is null || !context.Results.Any())
+            {
+                return false;
+            }
+
+            var outputFileName = $"UpgradeReport.{_options.Value.Format}";
```

Comment:
How to deal with conflicts? I thought there was logic for that in the analysis stuff....

Reply:
Good point, but since this change will also affects the analyze command, I'd rather make a new issue out of it and resolve it separately

Reply:
> How to deal with conflicts? I thought there was logic for that in the analysis stuff....

I don't believe there is - it stomps over old ones.